core/chains/evm/types/address.go

```diff
@@ -0,0 +1,47 @@
+package types
+
+import (
+	"bytes"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	commontypes "github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+var _ commontypes.Hashable = &Address{}
+
+type Address struct {
+	commontypes.Hashable
+	nativeAddress common.Address
+}
```

Comment:
Done, as per we discussed offline

Reply:
It will be more efficient, because all the cases can reference the same data in memory as whatever type they need, rather than copying it in to a new instance.

Reply:
Can you check the new version now?

Reply:
yeah we had casting directly, but it refused to play nicely with the interface constraints hence the struct wrapping instead - at the very least it removed redefining methods